Give ConfigurationData smart dimension default (#79)
glotzerlab/gsd: doc/schema-hoomd.rst

    Number of dimensions in the simulation. Must be 2 or 3.

>     Note:

Reviewer: ```suggestion
    .. note::
```
This section is not processed by napoleon.

        When using `gsd.hoomd.Snapshot`, the object will try to intelligently default to a

>         dimension. If setting the box to a box where :math:`L_z = 0`, ``dimensions`` will default to

Reviewer: ```suggestion
        dimension. When setting a box with :math:`L_z = 0`, ``dimensions`` will default to
```
